src/OpenApi/src/Transformers/Implementations/OpenApiSchemaReferenceTransformer.cs

```diff
@@ -112,6 +112,13 @@ public Task TransformAsync(OpenApiDocument document, OpenApiDocumentTransformerC
             return new OpenApiSchema { Reference = new OpenApiReference { Type = ReferenceType.Schema, Id = schemaId?.ToString() } };
         }
 
+        // Handle relative schemas that don't point to the parent document but to another property in the same type.
+        // In this case, remove the reference and rely on the properties that have been resolved and copied by the OpenApiSchemaService.
+        if (schema.Reference is { Type: ReferenceType.Schema, Id: var id } && id.StartsWith("#/", StringComparison.Ordinal))
```

Comment:
We resolve the JSON pointer that this ref points to in the new code `OpenApiSchemaService`. We retain the `$ref` so that we can use it as a comparison shorthand in OpenApiSchemaComparer. Here, before we resolve all the schemas in the document, we remove the relative reference so that the schema can be inlined using the properties that were copied over.